DfE.Hexagonal.Driver.FunctionApp/Triggers/CreateSchool/CreateSchoolFunction.cs

```diff
@@ -0,0 +1,59 @@
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using DfE.Hexagonal.Application.UseCases.School.CreateSchool;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+
+namespace DfE.Hexagonal.FunctionApp.Triggers.CreateSchool
+{
+    public class CreateSchoolFunction
+    {
+        private readonly ILogger<CreateSchoolFunction> _logger;
+        private readonly ICreateSchool _createSchool;
+        public CreateSchoolFunction(
+            ILogger<CreateSchoolFunction> log,
+            ICreateSchool createSchool)
+        {
+            _logger = log;
+            _createSchool = createSchool;
+        }
+
+        [FunctionName("CreateSchoolFunction")]
+        [OpenApiOperation(operationId: "create-school",
+            tags: new[] { "Create new School" },
+            Summary = "Creates a new school based on the details given",
+            Description = "Need to submit list of UPNs or searchText with optional filters, index type and query type",
+            Visibility = OpenApiVisibilityType.Important)]
+        [OpenApiRequestBody("input", typeof(SchoolInput))]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.Created, contentType: "text/plain", bodyType: typeof(SchoolInput), Description = "The details of the school created")]
+        [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.BadRequest,
+                                   Summary = "Invalid parameters",
+                                   Description = "Either Invalid parameters or parameters are missing")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = Routes.School)] HttpRequest req)
+        {
+            _logger.LogInformation("CreateSchoolFunction processed a request.");
+
+            string input = await new StreamReader(req.Body).ReadToEndAsync();
+            var schoolInput = JsonConvert.DeserializeObject<SchoolInput>(input);
+
+            var response = await _createSchool.Handle(schoolInput);
+            if (response.Success)
+            {
+                return new CreatedResult("", schoolInput);
+            }
+            else
+            {
+                return new BadRequestObjectResult(response.Errors);
+            }
+        }
+    }
+}
```

Comment:
Should we always have a trailing newline at the end of every file?

My personal view is yes but, ultimately, my preference is for consistency (always do it / never do it).

Do we have a team view? 

Once we have a view, could we programmatically enforce/encourage it using a `.editorconfig` file or similar, perhaps?

Reply:
Some coding expectations exist for an expectation of trailing new lines

On Github, it is flagged when no newline exists at the end of the file, even though it is cut from the visible code.

To save the mean looking red circle with a line in it, having a trailing newline would be nice.

Reply:
I actually didn't know for sure the rationale for this -- just force of habit on my part picked up from somewhere back in the day which I've never really questioned -- and turns out that it (mostly) dates back to the days of unix programs and things where, without a trailing newline, programs which ingest the files will behave oddly.

For example if we e.g. `cat` or `tail` the contents of a file to the console, then the beginning of the next part of terminal output would then either have to be exist on the trailing end of the last line of the file content or the terminal would need to insert a newline where it doesn't exist.

I see other mentions/references to the definition of a line literally being text content plus a newline character, which is interesting but possibly a bit academic/moot these days.

In practice and on Windows these days, it seems most programs will handle the absence of trailing newlines but including it is "more correct than not" -- e.g., in the case of GitHub, it recognises and explicitly flags the absence of a trailing newline as Scott noted (and this is what highlighted it to me). 

🤷